dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Intrinsics.java

```diff
@@ -400,6 +403,84 @@ private List<Event> inlineAtomicEnd(FunctionCall ignored) {
         return List.of(EventFactory.Svcomp.newEndAtomic(checkNotNull(currentAtomicBegin)));
     }
 
+    private final static FunctionType PTHREAD_THREAD_TYPE = types.getFunctionType(
+            types.getPointerType(), List.of(types.getPointerType())
+    );
+
+    private List<Event> inlinePthreadCreate(FunctionCall call) {
+        final List<Expression> arguments = call.getArguments();
+        assert arguments.size() == 4;
+        final Expression pidResultAddress = arguments.get(0);
+        //final Expression attributes = arguments.get(1);
+        final Expression targetFunction = arguments.get(2);
+        final Expression argument = arguments.get(3);
+
+        final Register resultRegister = getResultRegister(call);
+        assert resultRegister.getType() instanceof IntegerType;
+
+        final Register tidReg = call.getFunction().newUniqueRegister("__tid", types.getArchType());
+        final DynamicThreadCreate createEvent = newDynamicThreadCreate(tidReg, PTHREAD_THREAD_TYPE, targetFunction, List.of(argument));
+
+        return eventSequence(
+                createEvent,
+                newStore(pidResultAddress, tidReg),
+                // TODO: Allow to return failure value (!= 0)
+                newLocal(resultRegister, expressions.makeZero((IntegerType) resultRegister.getType()))
```

Comment:
This completely breaks `THREAD_CREATE_ALWAYS_SUCCEEDS` and needs to be fixed

Reply:
I'm not sure what you mean. Did you encounter an issue?
It would surprise me, because the semantics is identical to what it used to be: we always return success, even if the thread creation failed. Not only is the semantics identical, but the code we generate is literally the same.

Reply:
I did not encounter any problem, but I was worried about this https://github.com/hernanponcedeleon/Dat3M/blob/b3db2b01607b54a17c2fcd4e97cee5528eda96fb/dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java#L515

However, this seems to be dead code because we AFAIK we do not have events tagged by `Tag.STARTLOAD`. Shall we get rid of this and the option `THREAD_CREATE_ALWAYS_SUCCEEDS` (at least until we take care of this TODO)?

Reply:
That printing is certainly broken. I fixed it in #869. That being said, even in #869 we would return the success error code for spuriously failing pthread_create calls.